### ISSUE TITLE
centerpoint 模型跑nuscenes mini数据集loss为nan？

### ISSUE THREAD

**Issue by chenrui17** [PaddlePaddle/Paddle3D#61]
LAUNCH INFO 2022-08-29 11:48:40,554 devices: None
LAUNCH INFO 2022-08-29 11:48:40,554 elastic_level: -1
LAUNCH INFO 2022-08-29 11:48:40,554 elastic_timeout: 30
LAUNCH INFO 2022-08-29 11:48:40,555 gloo_port: 6767
LAUNCH INFO 2022-08-29 11:48:40,555 host: None
LAUNCH INFO 2022-08-29 11:48:40,555 job_id: default
LAUNCH INFO 2022-08-29 11:48:40,555 legacy: False
LAUNCH INFO 2022-08-29 11:48:40,555 log_dir: log
LAUNCH INFO 2022-08-29 11:48:40,555 log_level: INFO
LAUNCH INFO 2022-08-29 11:48:40,555 master: None
LAUNCH INFO 2022-08-29 11:48:40,555 max_restart: 3
LAUNCH INFO 2022-08-29 11:48:40,555 nnodes: 1
LAUNCH INFO 2022-08-29 11:48:40,555 nproc_per_node: None
LAUNCH INFO 2022-08-29 11:48:40,555 rank: -1
LAUNCH INFO 2022-08-29 11:48:40,555 run_mode: collective
LAUNCH INFO 2022-08-29 11:48:40,555 server_num: None
LAUNCH INFO 2022-08-29 11:48:40,556 servers:
LAUNCH INFO 2022-08-29 11:48:40,556 trainer_num: None
LAUNCH INFO 2022-08-29 11:48:40,556 trainers:
LAUNCH INFO 2022-08-29 11:48:40,556 training_script: 0,1,2,3
LAUNCH INFO 2022-08-29 11:48:40,556 training_script_args: ['tools/train.py', '--config', 'configs/centerpoint/centerpoint_pillars_02voxel_nuscenes_10sweep.yml', '--save_dir', './output_nuscenes', '--num_workers', '3', '--save_interval', '5']
LAUNCH INFO 2022-08-29 11:48:40,556 with_gloo: 0
LAUNCH INFO 2022-08-29 11:48:40,556 --------------------------------------------------
LAUNCH WARNING 2022-08-29 11:48:40,556 Compatible mode enable with args ['--gpus']
-----------  Configuration Arguments -----------
backend: auto
cluster_topo_path: None
elastic_pre_hook: None
elastic_server: None
enable_auto_mapping: False
force: False
gpus: 0,1,2,3
heter_devices:
heter_worker_num: None
heter_workers:
host: None
http_port: None
ips: 127.0.0.1
job_id: None
log_dir: log
np: None
nproc_per_node: None
rank_mapping_path: None
run_mode: None
scale: 0
server_num: None
servers:
training_script: tools/train.py
training_script_args: ['--config', 'configs/centerpoint/centerpoint_pillars_02voxel_nuscenes_10sweep.yml', '--save_dir', './output_nuscenes', '--num_workers', '3', '--save_interval', '5']
worker_num: None
WARNING 2022-08-29 11:48:40,560 launch.py:519] Not found distinct arguments and compiled with cuda or xpu or npu or mlu. Default use collective mode
WARNING 2022-08-29 11:48:40,560 launch.py:519] Not found distinct arguments and compiled with cuda or xpu or npu or mlu. Default use collective mode
launch train in GPU mode!
INFO 2022-08-29 11:48:40,562 launch_utils.py:679] Change selected_gpus into reletive values. --ips:0,1,2,3 will change into relative_ips:[0, 1, 2, 3] according to your CUDA_VISIBLE_DEVICES:['0', '1', '2', '3', '4', '5', '6', '7']
INFO 2022-08-29 11:48:40,562 launch_utils.py:679] Change selected_gpus into reletive values. --ips:0,1,2,3 will change into relative_ips:[0, 1, 2, 3] according to your CUDA_VISIBLE_DEVICES:['0', '1', '2', '3', '4', '5', '6', '7']
INFO 2022-08-29 11:48:40,565 launch_utils.py:562] Local start 4 processes. First process distributed environment info (Only For Debug):
    +=======================================================================================+
    |                        Distributed Envs                      Value                    |
    +---------------------------------------------------------------------------------------+
    |                       PADDLE_TRAINER_ID                        0                      |
    |                 PADDLE_CURRENT_ENDPOINT                 127.0.0.1:25055               |
    |                     PADDLE_TRAINERS_NUM                        4                      |
    |                PADDLE_TRAINER_ENDPOINTS  ... 0.1:22795,127.0.0.1:13246,127.0.0.1:10031|
    |                     PADDLE_RANK_IN_NODE                        0                      |
    |                 PADDLE_LOCAL_DEVICE_IDS                        0                      |
    |                 PADDLE_WORLD_DEVICE_IDS                     0,1,2,3                   |
    |                     FLAGS_selected_gpus                        0                      |
    |             FLAGS_selected_accelerators                        0                      |
    +=======================================================================================+

INFO 2022-08-29 11:48:40,565 launch_utils.py:562] Local start 4 processes. First process distributed environment info (Only For Debug):
    +=======================================================================================+
    |                        Distributed Envs                      Value                    |
    +---------------------------------------------------------------------------------------+
    |                       PADDLE_TRAINER_ID                        0                      |
    |                 PADDLE_CURRENT_ENDPOINT                 127.0.0.1:25055               |
    |                     PADDLE_TRAINERS_NUM                        4                      |
    |                PADDLE_TRAINER_ENDPOINTS  ... 0.1:22795,127.0.0.1:13246,127.0.0.1:10031|
    |                     PADDLE_RANK_IN_NODE                        0                      |
    |                 PADDLE_LOCAL_DEVICE_IDS                        0                      |
    |                 PADDLE_WORLD_DEVICE_IDS                     0,1,2,3                   |
    |                     FLAGS_selected_gpus                        0                      |
    |             FLAGS_selected_accelerators                        0                      |
    +=======================================================================================+

INFO 2022-08-29 11:48:40,565 launch_utils.py:566] details about PADDLE_TRAINER_ENDPOINTS can be found in log/endpoints.log, and detail running logs maybe found in log/workerlog.0
INFO 2022-08-29 11:48:40,565 launch_utils.py:566] details about PADDLE_TRAINER_ENDPOINTS can be found in log/endpoints.log, and detail running logs maybe found in log/workerlog.0
launch proc_id:21417 idx:0
launch proc_id:21436 idx:1
launch proc_id:21455 idx:2
launch proc_id:21478 idx:3
2022-08-29 11:48:45,357 -     INFO - Load 2554 pedestrian database infos
2022-08-29 11:48:45,358 -     INFO - Load 3035 car database infos
2022-08-29 11:48:45,358 -     INFO - Load 771 traffic_cone database infos
2022-08-29 11:48:45,358 -     INFO - Load 129 bicycle database infos
2022-08-29 11:48:45,358 -     INFO - Load 1340 barrier database infos
2022-08-29 11:48:45,359 -     INFO - Load 356 truck database infos
2022-08-29 11:48:45,359 -     INFO - Load 243 bus database infos
2022-08-29 11:48:45,359 -     INFO - Load 168 construction_vehicle database infos
2022-08-29 11:48:45,359 -     INFO - Load 159 motorcycle database infos
2022-08-29 11:48:45,359 -     INFO - Load 18 trailer database infos
2022-08-29 11:48:45,359 -     INFO - After filtering min_num_points_in_box:
2022-08-29 11:48:45,359 -     INFO - Load 2402 pedestrian database infos
2022-08-29 11:48:45,359 -     INFO - Load 2756 car database infos
2022-08-29 11:48:45,359 -     INFO - Load 639 traffic_cone database infos
2022-08-29 11:48:45,359 -     INFO - Load 122 bicycle database infos
2022-08-29 11:48:45,359 -     INFO - Load 1293 barrier database infos
2022-08-29 11:48:45,359 -     INFO - Load 334 truck database infos
2022-08-29 11:48:45,359 -     INFO - Load 226 bus database infos
2022-08-29 11:48:45,359 -     INFO - Load 161 construction_vehicle database infos
2022-08-29 11:48:45,359 -     INFO - Load 133 motorcycle database infos
2022-08-29 11:48:45,359 -     INFO - Load 16 trailer database infos
2022-08-29 11:48:45,360 -     INFO - Loading nuscenes metadata, this may take a few minutes
platform:
    Linux-4.14.0_1-0-0-44-x86_64-with-debian-bullseye-sid
    gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
    Python - 3.6.15 | packaged by conda-forge | (default, Dec  3 2021, 18:49:41)  [GCC 9.4.0]

Science Toolkits:
    cv2 - 4.6.0
    numpy - 1.19.5
    numba - 0.53.1
    pandas - 1.1.5
    pillow - 8.3.2
    skimage - 0.17.2

PaddlePaddle:
    paddle(gpu) - 2.3.2
    paddle3d - 0.5.0
    paddleseg - 2.6.0
    FLAGS_cudnn_deterministic - Not set.
    FLAGS_cudnn_exhaustive_search - Not set.

CUDA:
    cudnn - 8400
    nvcc - Build cuda_11.7.r11.7/compiler.31294372_0

GPUs:
    Not Found.
------------------------------------------------
2022-08-29 11:48:46,177 -     INFO -
---------------Config Information---------------
batch_size: 4
epochs: 20
lr_scheduler:
  base_learning_rate: 0.0001
  lr_ratio_peak: 10
  lr_ratio_trough: 0.0001
  step_ratio_peak: 0.4
  type: OneCycleWarmupDecayLr
model:
  backbone:
    downsample_strides:
    - 2
    - 2
    - 2
    in_channels: 64
    layer_nums:
    - 3
    - 5
    - 5
    out_channels:
    - 64
    - 128
    - 256
    type: SecondBackbone
  bbox_head:
    code_weights:
    - 1.0
    - 1.0
    - 1.0
    - 1.0
    - 1.0
    - 1.0
    - 0.2
    - 0.2
    - 1.0
    - 1.0
    common_heads:
      dim:
      - 3
      - 2
      height:
      - 1
      - 2
      reg:
      - 2
      - 2
      rot:
      - 2
      - 2
      vel:
      - 2
      - 2
    in_channels: 384
    tasks:
    - class_names:
      - car
      num_class: 1
    - class_names:
      - truck
      - construction_vehicle
      num_class: 2
    - class_names:
      - bus
      - trailer
      num_class: 2
    - class_names:
      - barrier
      num_class: 1
    - class_names:
      - motorcycle
      - bicycle
      num_class: 2
    - class_names:
      - pedestrian
      - traffic_cone
      num_class: 2
    type: CenterHead
    weight: 0.25
  middle_encoder:
    in_channels: 64
    point_cloud_range:
    - -51.2
    - -51.2
    - -5.0
    - 51.2
    - 51.2
    - 3.0
    type: PointPillarsScatter
    voxel_size:
    - 0.2
    - 0.2
    - 8
  neck:
    in_channels:
    - 64
    - 128
    - 256
    out_channels:
    - 128
    - 128
    - 128
    type: SecondFPN
    upsample_strides:
    - 0.5
    - 1
    - 2
    use_conv_for_no_stride: true
  test_cfg:
    down_ratio: 4
    max_per_img: 500
    nms:
      nms_iou_threshold: 0.2
      nms_post_max_size: 83
      nms_pre_max_size: 1000
    point_cloud_range:
    - -51.2
    - -51.2
    post_center_limit_range:
    - -61.2
    - -61.2
    - -10.0
    - 61.2
    - 61.2
    - 10.0
    score_threshold: 0.1
    voxel_size:
    - 0.2
    - 0.2
  type: CenterPoint
  voxel_encoder:
    feat_channels:
    - 64
    - 64
    in_channels: 5
    legacy: false
    max_num_points_in_voxel: 20
    point_cloud_range:
    - -51.2
    - -51.2
    - -5.0
    - 51.2
    - 51.2
    - 3.0
    type: PillarFeatureNet
    voxel_size:
    - 0.2
    - 0.2
    - 8
    with_distance: false
  voxelizer:
    max_num_points_in_voxel: 20
    max_num_voxels:
    - 30000
    - 60000
    point_cloud_range:
    - -51.2
    - -51.2
    - -5.0
    - 51.2
    - 51.2
    - 3.0
    type: HardVoxelizer
    voxel_size:
    - 0.2
    - 0.2
    - 8
optimizer:
  beta1:
    momentum_peak: 0.95
    momentum_trough: 0.85
    step_ratio_peak: 0.4
    type: OneCycleDecayWarmupMomentum
  beta2: 0.99
  grad_clip:
    clip_norm: 35
    type: ClipGradByGlobalNorm
  type: OneCycleAdam
  weight_decay: 0.01
train_dataset:
  class_balanced_sampling: true
  class_names:
  - car
  - truck
  - construction_vehicle
  - bus
  - trailer
  - barrier
  - motorcycle
  - bicycle
  - pedestrian
  - traffic_cone
  dataset_root: datasets/nuscenes/
  max_sweeps: 10
  mode: train
  transforms:
  - dim: 5
    sweep_remove_radius: 1
    type: LoadPointCloud
    use_dim: 4
    use_time_lag: true
  - class_names:
    - car
    - truck
    - construction_vehicle
    - bus
    - trailer
    - barrier
    - motorcycle
    - bicycle
    - pedestrian
    - traffic_cone
    database_anno_path: datasets/nuscenes/gt_database_train_nsweeps10_withvelo/anno_info_train_nsweeps10_withvelo.pkl
    database_root: datasets/nuscenes/
    max_num_samples_per_class:
      barrier: 2
      bicycle: 6
      bus: 4
      car: 2
      construction_vehicle: 7
      motorcycle: 6
      pedestrian: 2
      traffic_cone: 2
      trailer: 6
      truck: 3
    min_num_points_in_box_per_class:
      barrier: 5
      bicycle: 5
      bus: 5
      car: 5
      construction_vehicle: 5
      motorcycle: 5
      pedestrian: 5
      traffic_cone: 5
      trailer: 5
      truck: 5
    type: SamplingDatabase
  - type: RandomVerticalFlip
  - type: RandomHorizontalFlip
  - max_rot: 0.3925
    min_rot: -0.3925
    type: GlobalRotate
  - max_scale: 1.05
    min_scale: 0.95
    type: GlobalScale
  - type: ShufflePoint
  - point_cloud_range:
    - -51.2
    - -51.2
    - -5.0
    - 51.2
    - 51.2
    - 3.0
    type: FilterBBoxOutsideRange
  - down_ratio: 4
    gaussian_overlap: 0.1
    max_objs: 500
    min_radius: 2
    point_cloud_range:
    - -51.2
    - -51.2
    - -5.0
    - 51.2
    - 51.2
    - 3.0
    tasks:
    - class_names:
      - car
      num_class: 1
    - class_names:
      - truck
      - construction_vehicle
      num_class: 2
    - class_names:
      - bus
      - trailer
      num_class: 2
    - class_names:
      - barrier
      num_class: 1
    - class_names:
      - motorcycle
      - bicycle
      num_class: 2
    - class_names:
      - pedestrian
      - traffic_cone
      num_class: 2
    type: Gt2CenterPointTarget
    voxel_size:
    - 0.2
    - 0.2
    - 8
  type: NuscenesPCDataset
val_dataset:
  class_names:
  - car
  - truck
  - construction_vehicle
  - bus
  - trailer
  - barrier
  - motorcycle
  - bicycle
  - pedestrian
  - traffic_cone
  dataset_root: datasets/nuscenes/
  max_sweeps: 10
  mode: val
  transforms:
  - dim: 5
    sweep_remove_radius: 1
    type: LoadPointCloud
    use_dim: 4
    use_time_lag: true
  type: NuscenesPCDataset
------------------------------------------------
W0829 11:48:46.179867 21417 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 11.7, Runtime API Version: 11.2
W0829 11:48:46.179919 21417 gpu_resources.cc:91] device: 0, cuDNN Version: 8.4.
2022-08-29 11:48:48,264 -     INFO - Finish CenterHead Initialization
server not ready, wait 3 sec to retry...
not ready endpoints:['127.0.0.1:22795', '127.0.0.1:13246', '127.0.0.1:10031']
I0829 11:48:51.603751 21417 nccl_context.cc:83] init nccl context nranks: 4 local rank: 0 gpu id: 0 ring id: 0
2022-08-29 11:48:59,906 -  WARNING - No custom op voxelize found, try JIT build
Compiling user custom op, it will cost a few seconds.....
2022-08-29 11:49:10,431 -     INFO - voxelize builded success!
2022-08-29 11:49:24,364 -     INFO - [TRAIN] epoch=1/20, iter=10/1540, loss=nan, lr=0.000100 | ETA 00:17:18
2022-08-29 11:49:30,670 -     INFO - [TRAIN] epoch=1/20, iter=20/1540, loss=nan, lr=0.000102 | ETA 00:16:22
2022-08-29 11:49:37,130 -     INFO - [TRAIN] epoch=1/20, iter=30/1540, loss=nan, lr=0.000105 | ETA 00:16:05
2022-08-29 11:49:43,530 -     INFO - [TRAIN] epoch=1/20, iter=40/1540, loss=nan, lr=0.000108 | ETA 00:16:47
2022-08-29 11:49:49,571 -     INFO - [TRAIN] epoch=1/20, iter=50/1540, loss=nan, lr=0.000113 | ETA 00:14:50
2022-08-29 11:49:55,944 -     INFO - [TRAIN] epoch=1/20, iter=60/1540, loss=nan, lr=0.000120 | ETA 00:15:43
2022-08-29 11:50:02,244 -     INFO - [TRAIN] epoch=1/20, iter=70/1540, loss=nan, lr=0.000127 | ETA 00:14:39
2022-08-29 11:50:14,494 -     INFO - [TRAIN] epoch=2/20, iter=80/1540, loss=nan, lr=0.000135 | ETA 00:32:34
2022-08-29 11:50:21,012 -     INFO - [TRAIN] epoch=2/20, iter=90/1540, loss=nan, lr=0.000145 | ETA 00:16:39
2022-08-29 11:50:27,129 -     INFO - [TRAIN] epoch=2/20, iter=100/1540, loss=nan, lr=0.000155 | ETA 00:14:05
2022-08-29 11:50:33,403 -     INFO - [TRAIN] epoch=2/20, iter=110/1540, loss=nan, lr=0.000167 | ETA 00:15:34
2022-08-29 11:50:39,488 -     INFO - [TRAIN] epoch=2/20, iter=120/1540, loss=nan, lr=0.000179 | ETA 00:14:25
2022-08-29 11:50:45,807 -     INFO - [TRAIN] epoch=2/20, iter=130/1540, loss=nan, lr=0.000193 | ETA 00:14:04

**Comment by FlyingQianMM** [comment]
loss为nan由nuscenes-devkit造成，需要卸载掉pip安装的nuscenes-devkit，通过如下方式进行安装：
```
git clone https://github.com/tianweiy/nuscenes-devkit
export PYTHONPATH="${PYTHONPATH}:PATH_TO_NUSCENES_DEVKIT/python-sdk"
```

nuscenes-devkit develop与已发版的diff目前还在查。

**Comment by chenrui17** [comment]
> ${PYTHONPATH}:

GOOD！已解决
